### PR TITLE
validation in web.app

### PR DESCRIPTION
**Pull request by brtkkndrl**
validation checks for:
- question text
- user name
- quiz name

creation forms in question and answer list pages, are now hidden if not loaded

complex validations like
- quiz start < quiz end
- Time.now < quiz start
- answer text != null || answer picure != null

Do we need those or should it be left to the API? It would require custom validation on form or on model.

### CODE REVIEW

**Review by KoubaFromPucks — APPROVED** [review]

File: src/Quizzer.Web.App/Pages/QuestionList.razor [hunk 1]
```diff
@@ -14,27 +14,33 @@
...
-		<button class="btn btn-secondary" @onclick="@LoadQuizzes">load quizzes</button>
-	</div>
-	<div class="col-sm">
-		<label class="form-check-label" for="quizSelect">choose quiz:</label>
-	</div>
-	<div class="col-sm">
+	@if (_quizzes is null)
+	{
+		<div class="col-sm">
+			<button class="btn btn-secondary" @onclick="@LoadQuizzes">creation form</button>
```
Comment (vecerap) [inline src/Quizzer.Web.App/Pages/QuestionList.razor]: It will look weird when buttons does not have capitalization

**Comment by vecerap** [comment]
> validation checks for:
> 
> * question text
> * user name
> * quiz name
> 
> creation forms in question and answer list pages, are now hidden if not loaded
> 
> complex validations like
> 
> * quiz start < quiz end
> * Time.now < quiz start
> * answer text != null || answer picure != null
> 
> Do we need those or should it be left to the API? It would require custom validation on form or on model.

@BRTEK007 aren't those validations done through model attributes?

File: src/Quizzer.Web.App/Pages/QuestionList.razor (hunk 1 above)
Comment (brtkkndrl) [reply src/Quizzer.Web.App/Pages/QuestionList.razor]: changed

**Comment by brtkkndrl** [comment]
> > validation checks for:
> > 
> > * question text
> > * user name
> > * quiz name
> > 
> > creation forms in question and answer list pages, are now hidden if not loaded
> > complex validations like
> > 
> > * quiz start < quiz end
> > * Time.now < quiz start
> > * answer text != null || answer picure != null
> > 
> > Do we need those or should it be left to the API? It would require custom validation on form or on model.
> 
> @BRTEK007 aren't those validations done through model attributes?

yes they can be, but you need custom class for more complex validations.
I added it for QuizDetailModel, and AnswerDetailModel, also added hiding message box on invalid submit, and showing validation summary because EditForm hides error messages in my browser, if you spam click.

**Comment by vecerap** [comment]
Squash

**Review by vecerap — APPROVED** [review]

**Pull request merged**